lightning-background-processor/src/lib.rs

```diff
@@ -1175,41 +1361,111 @@ mod tests {
-			let p2p_gossip_sync = Arc::new(P2PGossipSync::new(network_graph.clone(), Some(chain_source.clone()), logger.clone()));
-			let rapid_gossip_sync = Arc::new(RapidGossipSync::new(network_graph.clone(), logger.clone()));
+			let manager = Arc::new(ChannelManager::new(
+				fee_estimator.clone(),
+				chain_monitor.clone(),
+				tx_broadcaster.clone(),
+				router.clone(),
+				logger.clone(),
+				keys_manager.clone(),
+				keys_manager.clone(),
+				keys_manager.clone(),
+				UserConfig::default(),
+				params,
+				genesis_block.header.time,
+			));
+			let p2p_gossip_sync = Arc::new(P2PGossipSync::new(
+				network_graph.clone(),
+				Some(chain_source.clone()),
+				logger.clone(),
+			));
+			let rapid_gossip_sync =
+				Arc::new(RapidGossipSync::new(network_graph.clone(), logger.clone()));
 			let msg_handler = MessageHandler {
-				chan_handler: Arc::new(test_utils::TestChannelMessageHandler::new(ChainHash::using_genesis_block(Network::Testnet))),
+				chan_handler: Arc::new(test_utils::TestChannelMessageHandler::new(
+					ChainHash::using_genesis_block(Network::Testnet),
+				)),
 				route_handler: Arc::new(test_utils::TestRoutingMessageHandler::new()),
-				onion_message_handler: IgnoringMessageHandler{}, custom_message_handler: IgnoringMessageHandler{}
+				onion_message_handler: IgnoringMessageHandler {},
+				custom_message_handler: IgnoringMessageHandler {},
+			};
+			let peer_manager = Arc::new(PeerManager::new(
+				msg_handler,
+				0,
+				&seed,
+				logger.clone(),
+				keys_manager.clone(),
+			));
+			let node = Node {
+				node: manager,
+				p2p_gossip_sync,
+				rapid_gossip_sync,
+				peer_manager,
+				chain_monitor,
+				kv_store,
+				tx_broadcaster,
+				network_graph,
+				logger,
+				best_block,
+				scorer,
 			};
-			let peer_manager = Arc::new(PeerManager::new(msg_handler, 0, &seed, logger.clone(), keys_manager.clone()));
-			let node = Node { node: manager, p2p_gossip_sync, rapid_gossip_sync, peer_manager, chain_monitor, kv_store, tx_broadcaster, network_graph, logger, best_block, scorer };
 			nodes.push(node);
 		}
 
 		for i in 0..num_nodes {
-			for j in (i+1)..num_nodes {
-				nodes[i].node.peer_connected(&nodes[j].node.get_our_node_id(), &Init {
-					features: nodes[j].node.init_features(), networks: None, remote_network_address: None
-				}, true).unwrap();
-				nodes[j].node.peer_connected(&nodes[i].node.get_our_node_id(), &Init {
-					features: nodes[i].node.init_features(), networks: None, remote_network_address: None
-				}, false).unwrap();
+			for j in (i + 1)..num_nodes {
+				nodes[i]
+					.node
+					.peer_connected(
+						&nodes[j].node.get_our_node_id(),
+						&Init {
+							features: nodes[j].node.init_features(),
+							networks: None,
+							remote_network_address: None,
+						},
```

Comment:
Some of these would probably be better formatted if we extract something like this into a local variable. Makes me wonder if we should do a file-by-file transition to catch these.